clang/runtime/dpct-rt/include/kernel.hpp.inc

```diff
@@ -62,13 +71,44 @@ struct kernel_function_info {
 };
 // DPCT_LABEL_END
 
+class kernel_function {
+public:
+  kernel_function()                         : ptr{nullptr} {}
+  kernel_function(dpct::kernel_functor ptr) : ptr{ptr}     {}
+
+  operator void *() const { return ((void *)ptr); }
+
+  void operator()(sycl::queue &q, const sycl::nd_range<3> &range,
+                  unsigned int a, void **args, void **extra) {
+    ptr(q,range,a,args,extra);
+  }
+
+private:
+  dpct::kernel_functor ptr;
+};
+
 // DPCT_LABEL_BEGIN|get_kernel_function_info|dpct
 // DPCT_DEPENDENCY_BEGIN
 // Kernel|kernel_function_info
 // Device|dev_mgr
 // Device|dev_mgr_current_device
 // DPCT_DEPENDENCY_END
 // DPCT_CODE
+static void get_kernel_function_info(kernel_function_info *kernel_info,
+                                     const kernel_function &function) {
+  kernel_info->max_work_group_size =
+      dpct::dev_mgr::instance()
+          .current_device()
+          .get_info<sycl::info::device::max_work_group_size>();
+}
+static kernel_function_info get_kernel_function_info(const kernel_function &function) {
+  kernel_function_info kernel_info;
```

Comment:
Similar problem to above function.
"static kernel_function_info get_kernel_function_info(const void *function)" should work.

Reply:
Removed unnecessary overloading.